compiler-core/src/erlang.rs

```diff
@@ -162,12 +162,26 @@ fn module_document<'a>(
         .append(line());
 
     for s in &module.definitions {
+        let mut export_anyway = false;
+
+        if module.definitions.iter().any(|def| match def {
```

Comment:
This traverses the AST of the module once per , which is `O(len(constants) * len(definitions))` additional work.

If instead we build a set of these private referenced modules when inferring constant expressions it'll be no additional work, let's do that.